From: Alex Tabachnik <alext@mellanox.com>
Subject: [PATCH] BACKPORT: ib_iser: function iser_attr_is_visible fix compilation warning incompatible type return value

This patch in general should fix patch backports/0022-BACKPORT-iser-for-SLES11SP1.patch
return value of function iser_attr_is_visible should be umode_t for kernels bigger then 3.3
and mode_t for all the RH and SLES distributions

issue: 182534
Change-Id: I28bab701399d6a13e8492afb66e8dc1bb15b91c4
Signed-off-by: Alex Tabachnik <alext@mellanox.com>
---
 drivers/infiniband/ulp/iser/iscsi_iser.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/infiniband/ulp/iser/iscsi_iser.c b/drivers/infiniband/ulp/iser/iscsi_iser.c
index xxxxxxx..xxxxxxx xxxxxx
--- a/drivers/infiniband/ulp/iser/iscsi_iser.c
+++ b/drivers/infiniband/ulp/iser/iscsi_iser.c
@@ -630,9 +630,13 @@ iscsi_iser_ep_disconnect(struct iscsi_endpoint *ep)
 	iser_err("ib conn %p state %d\n",ib_conn, ib_conn->state);
 	iser_conn_terminate(ib_conn);
 }
-
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)) || defined(CONFIG_COMPAT_ISER_ATTR_IS_VISIBLE)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0))
 static umode_t iser_attr_is_visible(int param_type, int param)
+#endif
+#if defined(CONFIG_COMPAT_ISER_ATTR_IS_VISIBLE)
+static mode_t iser_attr_is_visible(int param_type, int param)
+#endif
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,3,0)) || defined(CONFIG_COMPAT_ISER_ATTR_IS_VISIBLE)
 {
 	switch (param_type) {
 	case ISCSI_HOST_PARAM:
